py-polars/polars/expr/list.py

```diff
@@ -972,36 +997,58 @@ def to_struct(
             tracked or pre-determined, as the result will be eagerly evaluated,
             so you can leave this parameter unset.
 
+        Notes
+        -----
+        For performance reasons, the length of the first non-null sublist is used
+        to determine the number of output fields. If the sublists can be of different
+        lengths then `n_field_strategy="max_width"` must be used to obtain the expected
+        result.
+
         Examples
         --------
         Convert list to struct with default field name assignment:
 
-        >>> df = pl.DataFrame({"n": [[0, 1, 2], [0, 1]]})
-        >>> df.select(pl.col("n").list.to_struct())
-        shape: (2, 1)
-        ┌────────────┐
-        │ n          │
-        │ ---        │
-        │ struct[3]  │
-        ╞════════════╡
-        │ {0,1,2}    │
-        │ {0,1,null} │
-        └────────────┘
+        >>> df = pl.DataFrame({"n": [[0, 1], [0, 1, 2]]})
+        >>> df.with_columns(struct=pl.col("n").list.to_struct())
+        shape: (2, 2)
+        ┌───────────┬───────────┐
+        │ n         ┆ struct    │
+        │ ---       ┆ ---       │
+        │ list[i64] ┆ struct[2] │
+        ╞═══════════╪═══════════╡
+        │ [0, 1]    ┆ {0,1}     │
+        │ [0, 1, 2] ┆ {0,1}     │
+        └───────────┴───────────┘
+
+        In this case we must use the `max_width` strategy:
```

Comment:
Could you add a small inline explanation here as to *why* this case requires the `max_width` strategy? Then the casual reader doesn't have to go back up to the main docstring to work it out ✌️

Reply:
Good point, thank you. I've had a go at some improved wording and added comments the repr.

Reply:
Looks good 👌